dev/start_macros.h

```diff
@@ -14,3 +14,8 @@ __pragma(push_macro("min"))
 #endif  // defined(_MSC_VER)
 
 #include <ciso646>  //  due to #166
+
+#if __cplusplus == 201703L  // C++17 flag enabled
```

Comment:
oh really, good catch! I'll make the change so anybody using the -std:c++2a flag will still be able to use std::optional in the library.
Since I use a custom define (SQLITE_ORM_OPTIONAL_SUPPORTED), this adjustment will only touch a single line of code, if it may help taking a decision for your previous comment... :)